lib/matplotlib/lines.py

```diff
@@ -622,23 +622,29 @@ def recache(self, always=False):
         if len(x) != len(y):
             raise RuntimeError('xdata and ydata must be the same length')
 
-        x = x.reshape((len(x), 1))
-        y = y.reshape((len(y), 1))
+        self._xy = np.empty((len(x), 2), dtype=np.float_)
```

Comment:
This copy eliminates what I think is a less efficient copy in the `concatenate` calls.  There is no net change in the amount of copying.